### PR TITLE
Initial refactoring by @kuwagata

### PR DESCRIPTION
**Pull request by rackergs**
@Kuwagata what is the status of this refactor?

### CODE REVIEW

**Comment by Kuwagata** [comment]
@rackergs Hey Greg.  The client in its current state (in my branch) is functional.  When I wrote the refactor we were anticipating needing to talk to Atom Hopper, but due to a hiccup in our project's development I was forced to redirect my efforts elsewhere.  My plan was to add functional/unit tests (similar to the [RSE client](https://github.com/rackerlabs/python-rseclient) we wrote), but I did not have time to do so.

We'll still need to talk to Atom Hopper with this client, but I'm not sure what our timeframe is.  Are you planning on working on it soon?  I might be able to hack on it some during a two-day hack day coming up next week.  I may find other areas of improvement, since the code is over 6 months old now.

Tagging @BryanSD in case he's interested.

**Comment by gregswift** [comment]
I dont have any immediate plans, but i wasn't actively using the module either. I'd be interested in merging this PR and bumping the major version if you are.

**Comment by gregswift** [comment]
That being said i'd want to restore a few of the things your dropped, like the Makefile, .gitignore entries, and the spec file.

**Comment by BryanSD** [comment]
I don't really have much to add beyond what @Kuwagata already mentioned. I don't see any problems with keeping the Makefile, .gitignore entries, and the spec file.

**Comment by Kuwagata** [comment]
@gregswift Just merged your PR to my fork.

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,12 @@
 build/
-dist/
-rpms/
-rpm-build/
-MANIFEST
+artifacts/
+
+*.egg-info
 *.pyc
-*.pyo
```
Comment (gregswift) [inline .gitignore]: should we add pyo back in?

Comment (Kuwagata) [inline .gitignore]: Yeah no harm in that.  Adding that back in my local.

**Comment by gregswift** [comment]
I think i'm ready to merge this.  I did want to ask first...

So we've moved token handling completely into the hands of the consumers, yes?

**Comment by Kuwagata** [comment]
Yes, the intention was to make it more generic.

**Pull request merged**